core/src/main/java/org/aerogear/mobile/core/http/OkHttpResponse.java

```diff
@@ -25,9 +25,19 @@
     public OkHttpResponse(final Call okHttpCall, AppExecutors appExecutors) {
         appExecutors.networkThread().execute(() -> {
             try {
+                // this call will throw an exception only when a connection problem occurs
+                // even when there is a 400 or 500 no exception thrown
                 response = okHttpCall.execute();
                 requestCompleteLatch.countDown();
-                runSuccessHandler();
+
+                if(response.isSuccessful()){
+                    // status 200 or 300
+                    runSuccessHandler();
+                } else {
+                    // status 400 or 500
+                    error = new Exception(response.message());
```

Comment:
@danielpassos any better ideas here instead of plain Exception?

Reply:
What about raise a [HttpException](https://github.com/aerogear/aerogear-android-sdk/blob/master/core/src/main/java/org/aerogear/mobile/core/exception/HttpException.java) instead and update the HttpException to receive the Http status code on the constructor?

Reply:
ok, didn't notice we have that one. fixing it now.